### ISSUE TITLE
POST refset/members issues 

### ISSUE THREAD

**Issue by psiotwo** [b2ihealthcare/snow-owl#456]
Hi,

when trying to POST to refset/members, after a few subsequent posts the import fails with the the following exception. 

What would be the canonical way to create a SIMPLE_MAP refset and add members to it?

Thanks,

Petr

```
[2019-12-10T16:11:14.743] ERROR qtp329654759-46                                  com.b2international.snowowl.core.rest.ControllerExceptionMapper  Exception during request processing com.b2international.snowowl.core.api.SnowowlRuntimeException: java.lang.IllegalArgumentException: expected one element but was: <DatastoreLockIndexEntry{id=371, userId=snowowl, description=committing changes, parentDescription=<<root>>, repositoryId=snomedStore, branchPath=MAIN/SNOMEDCT-UK/BABYLON-97d972f8-4118-4b92-89a9-e9cf59a9cb11}, DatastoreLockIndexEntry{id=418, userId=snowowl, description=committing changes, parentDescription=<<root>>, repositoryId=snomedStore, branchPath=MAIN/SNOMEDCT-UK/BABYLON-97d972f8-4118-4b92-89a9-e9cf59a9cb11}>
	at com.b2international.snowowl.datastore.request.TransactionalRequest.execute(TransactionalRequest.java:66)
	at com.b2international.snowowl.datastore.request.TransactionalRequest.execute(TransactionalRequest.java:1)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.datastore.request.RevisionIndexReadRequest.lambda$0(RevisionIndexReadRequest.java:64)
	at com.b2international.index.revision.DefaultRevisionIndex.lambda$0(DefaultRevisionIndex.java:112)
	at com.b2international.index.DefaultIndex.read(DefaultIndex.java:36)
	at com.b2international.index.revision.DefaultRevisionIndex.read(DefaultRevisionIndex.java:112)
	at com.b2international.index.revision.DefaultRevisionIndex.read(DefaultRevisionIndex.java:106)
	at com.b2international.snowowl.datastore.request.RevisionIndexReadRequest.execute(RevisionIndexReadRequest.java:62)
	at com.b2international.snowowl.datastore.request.RevisionIndexReadRequest.execute(RevisionIndexReadRequest.java:1)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.datastore.request.BranchRequest.execute(BranchRequest.java:52)
	at com.b2international.snowowl.datastore.request.BranchRequest.execute(BranchRequest.java:1)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.datastore.request.IndexReadRequest.lambda$0(IndexReadRequest.java:45)
	at com.b2international.index.DefaultIndex.read(DefaultIndex.java:36)
	at com.b2international.snowowl.datastore.request.IndexReadRequest.execute(IndexReadRequest.java:43)
	at com.b2international.snowowl.datastore.request.IndexReadRequest.execute(IndexReadRequest.java:1)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.datastore.request.HealthCheckingRequest.execute(HealthCheckingRequest.java:47)
	at com.b2international.snowowl.datastore.request.HealthCheckingRequest.execute(HealthCheckingRequest.java:1)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.datastore.request.RepositoryRequest.execute(RepositoryRequest.java:62)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.core.rate.RateLimitingRequest.execute(RateLimitingRequest.java:50)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.core.authorization.AuthorizedRequest.execute(AuthorizedRequest.java:114)
	at com.b2international.snowowl.core.events.DelegatingRequest.next(DelegatingRequest.java:72)
	at com.b2international.snowowl.core.monitoring.MonitoredRequest.execute(MonitoredRequest.java:47)
	at com.b2international.snowowl.core.repository.ApiRequestHandler.handle(ApiRequestHandler.java:67)
	at com.b2international.snowowl.core.repository.ApiRequestHandler.handle(ApiRequestHandler.java:1)
	at com.b2international.snowowl.internal.eventbus.EventBus$1.run(EventBus.java:163)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: expected one element but was: <DatastoreLockIndexEntry{id=371, userId=snowowl, description=committing changes, parentDescription=<<root>>, repositoryId=snomedStore, branchPath=MAIN/SNOMEDCT-UK/BABYLON-97d972f8-4118-4b92-89a9-e9cf59a9cb11}, DatastoreLockIndexEntry{id=418, userId=snowowl, description=committing changes, parentDescription=<<root>>, repositoryId=snomedStore, branchPath=MAIN/SNOMEDCT-UK/BABYLON-97d972f8-4118-4b92-89a9-e9cf59a9cb11}>
	at com.google.common.collect.Iterators.getOnlyElement(Iterators.java:322)
	at com.google.common.collect.Iterators.getOnlyElement(Iterators.java:335)
	at com.google.common.collect.Iterables.getOnlyElement(Iterables.java:277)
	at com.b2international.snowowl.datastore.oplock.DatastoreOperationLockManager.getOrCreateLock(DatastoreOperationLockManager.java:297)
	at com.b2international.snowowl.datastore.oplock.DatastoreOperationLockManager.unlock(DatastoreOperationLockManager.java:149)
	at com.b2international.snowowl.datastore.oplock.DatastoreOperationLockManager.unlock(DatastoreOperationLockManager.java:125)
	at com.b2international.snowowl.core.repository.RepositoryTransactionContext.commit(RepositoryTransactionContext.java:252)
	at com.b2international.snowowl.datastore.request.TransactionalRequest.commit(TransactionalRequest.java:75)
	at com.b2international.snowowl.datastore.request.TransactionalRequest.execute(TransactionalRequest.java:62)
	... 36 common frames omitted
```

**Comment by cmark** [comment]
Hi @psiotwo,

Thanks for reporting. This is a known issue in the lock API, it will be fixed for the upcoming 7.2.0 release.

Cheers,
Mark

**Comment by cmark** [comment]
Hi @psiotwo,

This has been fixed and released within the 7.2.0 version. 
Feel free to reopen this ticket if the issue is still present in your system.
Closing.

Cheers,
Mark
